pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -163,33 +163,42 @@ func IsTrustedCABundleUpdated(ctx context.Context, cli client.Client, dscInit *d
 }
 
 func ConfigureTrustedCABundle(ctx context.Context, cli client.Client, log logr.Logger, dscInit *dsci.DSCInitialization, managementStateChanged bool) error {
-	switch dscInit.Spec.TrustedCABundle.ManagementState {
-	case operatorv1.Managed:
-		log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update cert configmaps")
-		istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
-		if err != nil {
-			return err
-		}
+	if dscInit.Spec.TrustedCABundle != nil {
+		switch dscInit.Spec.TrustedCABundle.ManagementState {
+		case operatorv1.Managed:
+			log.Info("Trusted CA Bundle injection is set to `Managed` state. Reconciling to add/update cert configmaps")
+			istrustedCABundleUpdated, err := IsTrustedCABundleUpdated(ctx, cli, dscInit)
+			if err != nil {
+				return err
+			}
 
-		if istrustedCABundleUpdated || managementStateChanged {
-			err = AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit)
+			if istrustedCABundleUpdated || managementStateChanged {
+				err = AddCABundleConfigMapInAllNamespaces(ctx, cli, dscInit)
+				if err != nil {
+					log.Error(err, "error adding configmap to all namespaces", "name", CAConfigMapName)
+					return err
```

Comment:
I think we should stick to the principle of handle or throw. In this case we do both - log (handle) and return (throw).

```suggestion
					fmt.Errorf("failed adding configmap %s to all namespaces: %w", CAConfigMapName, err)
```

Reply:
these all should have been changed to return fmt.Errorf(...)